### PR TITLE
Add Sale price validation

### LINKED ISSUE #37985: Product Block Editor: Add Sale price validation
**Description**

List price and Sale price fields have not been validated correctly

**AC**

- [ ] Both field are not required.
- [ ] Both field, if they have value, can't be lower than zero.
- [ ] List price can't be equals to or lower than sale price.
- [ ] Sale price can't be equals to or greater than list price.

### PR DESCRIPTION
**Pull request by mdperez86**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #37985

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Got to `/wp-admin/tools.php?page=woocommerce-admin-test-helper`
2. Under Features tab make sure to enable `product-block-editor`
3. Then visit `/wp-admin/admin.php?page=wc-admin&path=/add-product`
4. Under `General` tab and within `Basic details` section the product list price and sale price should validate according the AC described in #37985

### CODE REVIEW

File: packages/js/product-editor/src/blocks/regular-price/edit.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+		: null;
+
+	const regularPriceId = useInstanceId(
+		BaseControl,
+		'wp-block-woocommerce-product-regular-price-field'
+	) as string;
+
+	const regularPriceValidationError = useValidation(
+		'product/regular_price',
+		function nameValidator() {
```
Comment (joshuatf) [inline packages/js/product-editor/src/blocks/regular-price/edit.tsx]:
```suggestion
		function regularPriceValidator() {
```

File: packages/js/product-editor/src/blocks/sale-price/edit.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	} );
+
+	const salePriceId = useInstanceId(
+		BaseControl,
+		'wp-block-woocommerce-product-sale-price-field'
+	) as string;
+
+	const salePriceValidationError = useValidation(
+		'product/sale_price',
+		function nameValidator() {
```
Comment (joshuatf) [inline packages/js/product-editor/src/blocks/sale-price/edit.tsx]:
```suggestion
		function salePriceValidator() {
```

File: packages/js/product-editor/src/blocks/regular-price/edit.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+	);
+	const context = useContext( CurrencyContext );
+	const { getCurrencyConfig, formatAmount } = context;
+	const currencyConfig = getCurrencyConfig();
+	const inputProps = useCurrencyInputProps( {
+		value: regularPrice,
+		setValue: setRegularPrice,
+	} );
+
+	const interpolatedHelp = help
```
Comment (joshuatf) [inline packages/js/product-editor/src/blocks/regular-price/edit.tsx:56]: I don't see `help` currently being used.  Do we have plans to use this in the future?

**Review by joshuatf — COMMENTED** [review]
Code looks great!  I like the separation of these into two blocks.  Left a couple comments and a quick question on the `help` expectations.

Also, should we remove the old `woocommerce/product-pricing-field` block or is this still needed?

File: packages/js/product-editor/src/blocks/regular-price/edit.tsx (hunk 3 above)
Comment (mdperez86) [reply packages/js/product-editor/src/blocks/regular-price/edit.tsx:56]: Yes it is. See the next [line](https://github.com/woocommerce/woocommerce/pull/38078/files/179d419e93a7c7e5eedd1eaefc16b43f204662f3#diff-22543645991a77248897b85d0248931df6f38316cd7de38c2245972961b50316R57)

File: packages/js/product-editor/src/blocks/regular-price/edit.tsx (hunk 1 above)
Comment (mdperez86) [reply packages/js/product-editor/src/blocks/regular-price/edit.tsx]: Thanks. Done -> https://github.com/woocommerce/woocommerce/pull/38078/commits/ff88a412748311a0c84f0e9b23547a3bd1548e7a

File: packages/js/product-editor/src/blocks/sale-price/edit.tsx (hunk 2 above)
Comment (mdperez86) [reply packages/js/product-editor/src/blocks/sale-price/edit.tsx]: Done -> https://github.com/woocommerce/woocommerce/pull/38078/commits/ff88a412748311a0c84f0e9b23547a3bd1548e7a

File: packages/js/product-editor/src/blocks/regular-price/edit.tsx (hunk 3 above)
Comment (joshuatf) [reply packages/js/product-editor/src/blocks/regular-price/edit.tsx:56]: I can see it used in this component, but was curious if this was being actively passed as an attribute to this component.

I see now that a `help` attribute is being passed in [the PR ](https://github.com/woocommerce/woocommerce/pull/37961)merged in last week so this looks good. 👍

**Review by joshuatf — APPROVED** [review]
Sorry for the delay in follow-up review.  This LGTM!  🚢

**Pull request merged**